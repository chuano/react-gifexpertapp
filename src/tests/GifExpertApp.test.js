import '@testing-library/jest-dom'
import React from 'react'
import { shallow } from 'enzyme'
import { GifExpertApp } from '../GiftExpetApp'

describe('GifExpertApp', () => {
  test('debe mostrar el componente correctamente', () => {
    const wrapper = shallow(<GifExpertApp />)
    expect(wrapper).toMatchSnapshot()
  })
  
  test('debe mostrar una lista de categorÃ­as', () => {
    const categories = ['One punch', 'Dragon ball']
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })
})

