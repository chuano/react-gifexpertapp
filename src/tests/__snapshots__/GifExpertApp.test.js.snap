// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`GifExpertApp debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;
